mrmustard/physics/gaussian_integrals.py

```diff
@@ -149,9 +149,27 @@ def complex_gaussian_integral(
     return A_post, b_post, c_post
 
 
+def join_cs(
```

Comment:
Ok, perhaps instead of digging ourselves deeper into calling `outer` as a primitive operation, let's revisit why we need to call it. I think it's because we defined the gaussian integral as a function of a single tensor. If we had a version of `gaussian_integral` that takes two tensors then we wouldn't need to join them and call `outer` in the first place. What do you think?

Reply:
I agree. It is what we discussed, that it may be more efficient (at least in memory) to not call outer, but just store the two arrays that we would do the outer products on. 
The join_cs is just taking care of the three cases where both c are scalars, both are arrays, and one scalar one array. The question is if we need a separate `gaussian_integral `for each of these cases or we can make a solution that fits all, while not slowing down the "simple scalar" case.

Reply:
They are surely related, but not the same thing. This is much simpler. At the moment `join_cs` is just a step inside the inner product. But we don't need to take this outer product, and instead carry out the computation of the final result from the two parts. This requires modifying the way `gaussian_integral` takes arguments and works. It shouldn't be difficult, I'm happy to help.